scipy/spatial/ckdtree/src/query.cxx

```diff
@@ -381,6 +381,12 @@ query_single_point(const ckdtree *self,
                 npy_float64 tmp = x[split_dim] - split; 
                 if(NPY_LIKELY(p == 2)) {
                     inf2->side_distances()[split_dim] = tmp * tmp;
+                } else
+                if(NPY_LIKELY(p == 1)) {
+                    inf2->side_distances()[split_dim] = dabs(tmp);
+                } else
+                if(NPY_LIKELY(ckdtree_isinf(p))) {
```

Comment:
And I have a PR that makes the casings on ```p```  compile time constants, so it really doesn't matter. And besides we should not try to be smarter than the C++ compiler. In Python it matters, but this is C++.